application/comit_node/src/bam_api/mod.rs

```diff
@@ -40,9 +40,9 @@ impl ToBamHeader for LedgerKind {
 }
 
 impl FromBamHeader for SwapProtocol {
-    fn from_bam_header(header: Header) -> Result<Self, serde_json::Error> {
+    fn from_bam_header(mut header: Header) -> Result<Self, serde_json::Error> {
         Ok(match header.value::<String>()?.as_str() {
-            "COMIT-RFC-003" => SwapProtocol::Rfc003,
+            "COMIT-RFC-003" => SwapProtocol::Rfc003(header.take_parameter("hash_function")?),
```

Comment:
I don't know why it is uppercase, my Rust foo is non-existent at the moment so am unsure if this gets converted to lowercase at somestage.  In the end I left it as it was because of two reasons
1. I think it should be changed to `rfc003` to match other useage in the code base.
2. @da-kami suggested that perhaps we will find a better name all together than 'rfc003'.

I actually did number 1 as a previous patch when doing this version forgetting that we don't rebase PRs.  And since this change would require patching the RFC repo in parallel I decided the extra merge complexity was not worth the effort.

I can however add a patch to this PR to just lowercase `COMIT-RFC-003` if you there is consensus on that small change?

Reply:
I'd suggest we don't change it for now and re-align after discussion with a dedicated PR